src/Illuminate/Queue/SerializesModels.php

```diff
@@ -89,9 +93,22 @@ public function __unserialize(array $values)
                 continue;
             }
 
-            $property->setValue(
-                $this, $this->getRestoredPropertyValue($values[$name])
-            );
+            $value = $this->getRestoredPropertyValue($values[$name]);
+
+            $property->setValue($this, $value);
+
+            if (
+                ($value instanceof Model || $value instanceof Collection) &&
+                ($attribute = $property->getAttributes(EagerLoad::class)[0] ?? null)
+            ) {
+                $relations = $attribute->getArguments()[0];
+
+                $value->load($relations);
+            }
+        }
+
+        if (in_array(Queueable::class, class_uses_recursive($this)) && method_exists($this, 'initializeOnQueue')) {
```

Comment:
I feel like swapping the order would be more optimal, but probably not by much:

```php
if (method_exists($this, 'initializeOnQueue') && in_array(Queueable::class, class_uses_recursive($this))) {
```

Probably would only be important if this was called a lot.